src/mlpack/core/tree/cosine_tree/cosine_tree.hpp

```diff
@@ -169,7 +198,7 @@ class CosineTree
   void GetFinalBasis(arma::mat& finalBasis) { finalBasis = basis; }
 
   //! Get pointer to the dataset matrix.
-  const arma::mat& GetDataset() const { return dataset; }
+  arma::mat* GetDataset() { return dataset; }
```

Comment:
Let's leave this as a reference also so the external API for the class does not change.